p2p/switch.go

```diff
@@ -771,8 +740,17 @@ func (sw *Switch) addOutboundPeerWithConfig(
 		return err
 	}
 
-	nodeInfo, err := handshake(sw.nodeInfo, conn, sw.config.HandshakeTimeout)
+	stream, err := conn.OpenStream(HandshakeStreamID, nil)
```

Comment:
I wonder if this logic can be moved into the `handshake` method, since it is the same here and on the accept routine.